OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -116,12 +118,10 @@ void ITick.Tick(Actor self)
 
 		void SeedResources(Actor self)
 		{
-			var pieces = self.World.SharedRandom.Next(info.Pieces[0], info.Pieces[1]) * ticks / growTicks;
-			if (pieces < info.Pieces[0])
-				pieces = info.Pieces[0];
-
-			var cells = self.World.Map.FindTilesInAnnulus(self.Location, 1, info.Range).ToList();
-
+			var pieces = int2.Lerp(info.Bursts[0], info.Bursts[1], ticks, growTicks);
+			var range = int2.Lerp(info.Range[0], info.Range[1], ticks, growTicks);
+			var cells = self.World.Map.FindTilesInAnnulus(self.Location, 1, range).ToList();
```

Comment:
```suggestion
			var cells = self.World.Map.FindTilesInAnnulus(self.Location, 1, range).ToList();
			var emptyCells = cells
				.Where(p =>
					resourceLayer.GetResource(p).Type != info.ResourceType
					&& resourceLayer.CanAddResource(info.ResourceType, p))
				.ToList();
```

the current empty check doesn't work at all. It still fires onto terrain where spice can't be planted, if that's intentional then we should remove the check, if it's not then use my suggestion

Reply:
Lets go with your suggestion